ci/downstream.sh

```diff
@@ -0,0 +1,240 @@
+                rm -fr "${_build_dir}"
+            fi
+        fi
+    else
+        exit 0
+    fi
+}
+
+# Exit and handle cleanup processes if needed
+f_exit()
+{
+    f_cleanup
+    exit "$0"
+}
+
+f_create_collection_dir_structure()
+{
+    f_log_info "${FUNCNAME[0]}"
+    # Create the Collection
+    for f_name in "${_file_manifest[@]}";
+    do
+        cp "./${f_name}" "${_build_dir}/${f_name}"
+    done
+    for d_name in "${_dir_manifest[@]}";
+    do
+        cp -r "./${d_name}" "${_build_dir}/${d_name}"
+    done
+    for exclude_file in "${_file_exclude[@]}";
+    do
+        if [[ -f "${_build_dir}/${exclude_file}" ]]; then
+            rm -f "${_build_dir}/${exclude_file}"
+        fi
+    done
+}
+
+f_install_kubernetes_core_from_src()
+{
+    local community_k8s_tmpdir="${_tmp_dir}/community.kubernetes"
+    local install_collections_dir="${_tmp_dir}/ansible_collections"
+    mkdir -p "${community_k8s_tmpdir}"
+    pushd "${_tmp_dir}"
+    #    curl -L \
+    #        https://github.com/ansible-collections/community.kubernetes/archive/main.tar.gz \
+    #        | tar -xz -C "${community_k8s_tmpdir}" --strip-components 1
+    #    pushd "${community_k8s_tmpdir}"
+    #        make downstream-build
+    #        ansible-galaxy collection install -p "${install_collections_dir}" "${community_k8s_tmpdir}"/kubernetes-core-*.tar.gz
+    #    popd
+    #popd
+        git clone https://github.com/maxamillion/community.kubernetes "${community_k8s_tmpdir}"
+        pushd "${community_k8s_tmpdir}"
+            git checkout downstream/fix_collection_name
+            make downstream-build
+            ansible-galaxy collection install -p "${install_collections_dir}" "${community_k8s_tmpdir}"/kubernetes-core-*.tar.gz
+        popd
+    popd
+}
+
+
+f_copy_collection_to_working_dir()
+{
+    f_log_info "${FUNCNAME[0]}"
+    # Copy the Collection build result into original working dir
+    cp "${_build_dir}"/*.tar.gz ./
+}
+
+f_common_steps()
+{
+    f_log_info "${FUNCNAME[0]}"
+    f_prep
+    f_create_collection_dir_structure
+    f_text_sub
+}
+
+# Run the test sanity scanerio
+f_test_sanity_option()
+{
+    f_log_info "${FUNCNAME[0]}"
+    f_common_steps
+    f_install_kubernetes_core_from_src
+    pushd "${_build_dir}" || return
+        ansible-galaxy collection build
+        f_log_info "SANITY TEST PWD: ${PWD}"
+        ## Can't do this because the upstream community.kubernetes dependency logic
+        ## is bound as a Makefile dep to the test-sanity target
+        #SANITY_TEST_ARGS="--docker --color --python 3.6" make test-sanity
+        ansible-test sanity --docker -v --exclude ci/ --color --python 3.6
+    popd || return
+    f_cleanup
+}
+
+# Run the test integration
+f_test_integration_option()
+{
+    f_log_info "${FUNCNAME[0]}"
+    f_common_steps
+    f_install_kubernetes_core_from_src
+    pushd "${_build_dir}" || return
+        f_log_info "INTEGRATION TEST WD: ${PWD}"
+        make test-integration-incluster
```

Comment:
This will end up calling `make test-integration` in the image built from CI/Dockerfile, will that end up running anything from the downstream steps?